lib/vega_lite/data.ex

```diff
@@ -327,6 +327,381 @@ defmodule VegaLite.Data do
+
+  ## Options
+
+    * `:radius_marks` - optional list of values for the radius markings.
+    * `:direction` - one of `:counter_clockwise` or `:clockwise`, which will affect the direction
+      in which the angles grow from the starting point. Defaults to `:counter_clockwise`.
+    * `:angle_marks` - list of angles for marking the grid. Defaults to `Enum.to_list(0..360//90)`.
+    * `:angle_offset` - offset for the 0º reference. The sign will obey the `:direction` option. Defaults to `0`.
+    * `:opacity` - opacity of the grid. Defaults to `1`.
+    * `:color` - backround color for the grid. Defaults to `"white"`.
+    * `:stroke_color` - stroke color for the grid markings. Defaults to `"black"`.
+    * `:stroke_opacity` - opacity for the grid strokes. Defaults to `1`.
+
+  ## Examples
+
+  This first call will plot the grid using all default options.
+
+      VegaLite.Data.polar_grid([0, 1, 2, 3, 4, 5])
+
+  The second call is a more customized example, with custom angle offset, direction
+  and angle markings:
+
+      vl = VegaLite.new(title: "Polars", width: 500, height: 500)
+      VegaLite.Data.polar_grid(vl, [0, 1, 2, 3, 4, 5], angle_offset: -90, direction: :clockwise)
+  """
+  def polar_grid(radius_marks, opts \\ []), do: polar_grid(Vl.new(), radius_marks, opts)
+
+  def polar_grid(vl, radius_marks, opts) do
+    opts =
+      Keyword.validate!(
+        opts,
+        angle_marks: [0, 90, 180, 270, 360],
+        direction: :counter_clockwise,
+        angle_offset: 0,
+        opacity: 1,
+        color: "white",
+        stroke_color: "black",
+        stroke_opacity: 1
+      )
+
+    grid_layers = polar_angle_layers(opts)
+    radius_layers = polar_radius_layers(radius_marks, opts)
+
+    vl = append_layers(vl |> Vl.data_from_values(%{"_r" => [0]}), grid_layers ++ radius_layers)
+
+    put_in(
+      vl.spec["_vl_polar_config"],
+      opts
+      |> Keyword.take([:angle_offset, :direction])
+      |> Map.new()
+      |> Map.put(:radius_marks, radius_marks)
+      |> Map.put(:hide_axes, true)
+    )
+  end
+
+  @doc """
+  Receives data represented in the polar domain and plots it into a grid.
+
+  By default, the grid will be the usual cartesian grid, with the data converted
+  into rectangular coordinates. If the given `VegaLite` spec was generated through
+  `VegaLite.Data.polar_grid/3`, the cartesian grid will be hidden by default such
+  that the resulting plot is purely in the polar domain.
+
+  `data` contains data inside a container which supports the `Table.Reader` protocol.
+
+  `mark` is either an atom for the mark type (usually `:point` or `:line`) or a keyword
+  list containing the `:type` key specifying the mark type and any options for the mark.
+
+  The `fields` argument must contain specifications for the `:r` and `:theta`,
+  mapping them onto keys contained in `data`.
+
+  It may also contain the following meta-fields:
+
+    * `:x` - options for the x-axis
+    * `:y` - options for the y-axis
```

Comment:
```suggestion
```

Let's remove options for `:x` and `:y` for now. The most we can simplify this for now, the better.

Reply:
This will make it so we can't have the standalone cartesian-converted plot. This is what you want?

Reply:
Yes, we can wait for a use case.

Reply:
Done :)